### PR TITLE
hanko: init at 0.5.2

### PR DESCRIPTION
**Pull request by SRv6d**
Adds [hanko](https://github.com/SRv6d/hanko), a rust based tool that keeps git allowed signers up to date.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ha/hanko/package.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    installShellCompletion assets/completions/_hanko
+    installShellCompletion assets/completions/hanko.fish
+  '';
+
+  meta = with lib; {
+    description = "Keeps your Git allowed signers file up to date with signing keys configured on software development platforms like GitHub and GitLab";
+    homepage = "https://github.com/SRv6d/hanko";
+    changelog = "https://github.com/SRv6d/hanko/blob/main/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ SRv6d ];
```
Comment (Noodlez1232) [inline pkgs/by-name/ha/hanko/package.nix:30]: The case used is wrong. It should be `[ srv6d ]` as written in the maintainers.nix file.

File: pkgs/by-name/ha/hanko/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ lib, fetchFromGitHub, installShellFiles, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "hanko";
+  version = "0.5.2";
+
+  src = fetchFromGitHub {
+    owner = "SRv6d";
+    repo = pname;
```
Comment (pluiedev) [inline pkgs/by-name/ha/hanko/package.nix:9]: `repo = pname;` is an antipattern as the repo name isn't inherently related to the pname in any way
```suggestion
    repo = "hanko";
```

File: pkgs/by-name/ha/hanko/package.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    rev = "v${version}";
```
Comment (pluiedev) [inline pkgs/by-name/ha/hanko/package.nix:10]:
```suggestion
    tag = "v${version}";
```

File: pkgs/by-name/ha/hanko/package.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  postInstall = ''
+    installManPage assets/manpages/*.1
+
+    installShellCompletion assets/completions/hanko.bash
+    installShellCompletion assets/completions/_hanko
+    installShellCompletion assets/completions/hanko.fish
+  '';
+
+  meta = with lib; {
```
Comment (pluiedev) [inline pkgs/by-name/ha/hanko/package.nix:25]: Avoid with lib;
```suggestion
  meta = {
```

File: pkgs/by-name/ha/hanko/package.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    description = "Keeps your Git allowed signers file up to date with signing keys configured on software development platforms like GitHub and GitLab";
```
Comment (pluiedev) [inline pkgs/by-name/ha/hanko/package.nix:26]: Descriptions should be short, concise phrases that fit into one sentence. More description text could always be put into `longDescription` instead.

```suggestion
    description = "Keeps your Git allowed signers file up to date";
```

File: pkgs/by-name/ha/hanko/package.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{ lib, fetchFromGitHub, installShellFiles, rustPlatform }:
```
Comment (pluiedev) [inline pkgs/by-name/ha/hanko/package.nix:1]: Please reformat with `nixfmt-rfc-style`

**Review by pluiedev — CHANGES_REQUESTED** [review]
Welcome to Nixpkgs! Here's some suggestions:

File: pkgs/by-name/ha/hanko/package.nix (hunk 1 above)
Comment (SRv6d) [reply pkgs/by-name/ha/hanko/package.nix]: Thanks, fixed.

File: pkgs/by-name/ha/hanko/package.nix (hunk 4 above)
Comment (SRv6d) [reply pkgs/by-name/ha/hanko/package.nix:30]: I saw this commonly done in other package files, is there some sort of style guide that advocates against this?

Comment (pluiedev) [reply pkgs/by-name/ha/hanko/package.nix:30]: More detailed discussion can be found at https://github.com/NixOS/nixpkgs/issues/208242 — it's a style lint done for new packages but it hasn't been implemented for older packages yet

Comment (SRv6d) [reply pkgs/by-name/ha/hanko/package.nix]: My bad, didn't realize I wasn't actually using it at all, thanks for the link.

Comment (SRv6d) [reply pkgs/by-name/ha/hanko/package.nix]: *was needed but fixed it

**Review by pluiedev — APPROVED** [review]
LGTM, just make sure to squash the fixup commits into the main commits (we don't do squash merges in Nixpkgs)

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**